cni/network/network.go

```diff
@@ -446,123 +445,120 @@ func (plugin *NetPlugin) Add(args *cniSkel.CmdArgs) error {
 			return fmt.Errorf("%w", err)
 		}
 
-		ipamAddResult.ncResponse, ipamAddResult.hostSubnetPrefix, er = plugin.multitenancyClient.GetContainerNetworkConfiguration(
-			context.TODO(), nwCfg, k8sPodName, k8sNamespace)
-		if er != nil {
-			er = errors.Wrapf(er, "GetContainerNetworkConfiguration failed for podname %v namespace %v", k8sPodName, k8sNamespace)
-			log.Printf("%+v", er)
-			return er
+		ipamAddResults, err = plugin.multitenancyClient.GetAllNetworkContainers(context.TODO(), nwCfg, k8sPodName, k8sNamespace, args.IfName)
+		if err != nil {
+			err = errors.Wrapf(err, "GetAllNetworkContainers failed for podname %v namespace %v", k8sPodName, k8sNamespace)
+			log.Printf("%+v", err)
+			return err
 		}
-
-		ipamAddResult.ipv4Result = convertToCniResult(ipamAddResult.ncResponse, args.IfName)
-
-		log.Printf("PrimaryInterfaceIdentifier: %v", ipamAddResult.hostSubnetPrefix.IP.String())
+	} else {
+		// TODO: refactor this code for simplification
+		// Add empty ipamAddResult object for single tenancy mode
+		// this will be used for: ipamAddResult, err = plugin.ipamInvoker.Add(ipamAddConfig)
+		ipamAddResults = append(ipamAddResults, ipamAddResult)
```

Comment:
instead of this, we need to have better way.. how about having numNetworks variable with default value as 1 and then set to len(ipamAddResult) in case of mutltitenancy case and then use numNetworks to iterate

Reply:
let's name it as numEnpoints